### PR TITLE
generalizing cluster predictions with no test data

### PR DESCRIPTION
**Pull request by kzecchini**
`sklearn_model.py` and `sklearn_preprocessing_pipeline.py` - generalize to use `self` instead of class. This allows for parent classes to override the method with the same signature. Also allow models to find upstream data with no test set.

`sklearn_cluster_model.py` - update predictions. Before would inherit from `SklearnModel`, now implement own class. This is to allow for not using test data - if we want to cluster all train data. Instead of predictions, we return the attribute `self.model.labels_` which is available in all sklearn clustering models.

### CODE REVIEW

File: primrose/models/sklearn_cluster_model.py [hunk 1]
```diff
@@ -28,11 +28,69 @@ def fit_training_data(self):
...
+        if hasattr(self, 'predictions') and self.predictions is not None:
+            return
+
+        if self.model is None:
+            self.model = self.load_model(data_object)
+
+        self.X_train, self.y_train, self.X_test, self.y_test = self._get_data(data_object)
+
+        logging.info("Making predictions with model")
+        to_predict = self.X_test
```
Comment (briangrahamww) [inline primrose/models/sklearn_cluster_model.py]: maybe just move into the else statement to use the same pattern as the if statement
